src/main/java/RedisSessionManagerT8.java

```diff
@@ -0,0 +1,319 @@
+        Hashtable<String, Object> metadata = new Hashtable<>();
+        metadata.put(METADATA_VALID, String.valueOf(session.isValid()));
+        metadata.put(METADATA_CREATION_TIME, dateFormat.format(new Date(session.getCreationTime())));
+        metadata.put(METADATA_LAST_ACCESS_TIME, dateFormat.format(new Date(session.getLastAccessedTime())));
+        metadata.put(METADATA_MAX_INACTIVE_INTERVAL, String.valueOf(session.getMaxInactiveInterval()));
+        return metadata;
+    }
+
+    /**
+     * Get attributes from a Session
+     * @param session
+     * @return
+     */
+    private Hashtable<String, Object> getAttributes(Session session){
+        StandardSession standardSession = (StandardSession)session;
+        Hashtable<String, Object> attributes = new Hashtable();
+        for (Enumeration<String> enumerator = standardSession.getAttributeNames(); enumerator.hasMoreElements();) {
+            String key = enumerator.nextElement();
+            attributes.put(key, standardSession.getAttribute(key));
+        }
+        return attributes;
+    }
+
+    //------------------------------------------------------------------------------------------------------------------
+    // END Utils
+    //------------------------------------------------------------------------------------------------------------------
+
+
+    //------------------------------------------------------------------------------------------------------------------
+    // START REDIS
+    //------------------------------------------------------------------------------------------------------------------
+
+    /**
+     * Method to saved session in redis
+     * @param session
+     * @return
+     * @throws IOException
+     */
+    protected boolean saveSession(Session session) throws IOException {
+        boolean errorSaving = true;
+        //Get the session Id
+        String id = session.getId();
+        Hashtable<String, Object> metadata = getMetadata(session);
+        Hashtable<String, Object> attributes = getAttributes(session);
+        try{
+            //Save metadata to Redis
+            byte[] encodedMetadata = Base64.getEncoder().encode(SerializationUtils.serialize(metadata));
+            withRedis((Jedis jedis)-> jedis.set((id + REDIS_METADATA_KEY).getBytes(), encodedMetadata));
+
+            //Save Attributes to Redis
+            byte[] encodedAttributes = Base64.getEncoder().encode(SerializationUtils.serialize(attributes));
+            withRedis((Jedis jedis)-> jedis.setex((id + REDIS_ATTRIBUTES_KEY).getBytes(), session.getMaxInactiveInterval(), encodedAttributes));
+
+            errorSaving = false;
+        }catch (Exception e){
+            log.error("Error - Context: saveToRedis. Description: " + e.getMessage());
+        }
+        return errorSaving;
+    }
+
+    private interface RedisCallback<T> {
+        T apply(Jedis jedis) throws IOException;
+    }
+
+    private <T> T withRedis(RedisCallback<T> callback) throws IOException {
```

Comment:
Add redis transaction handling here: call `MULTI` at the beginning and `EXEC` at the end if successful, and `DISCARD` if there was a failure. http://redis.io/topics/transactions